### PR TITLE
SPU: Rewrite ADSR/Volume slides

### PR DESCRIPTION
**Pull request by Ziemas**
### Description of Changes
Rewrite ADSR and volume slides, also fully moves the SPU over to 16bit volumes getting rid of MulShr32 etc.

### Rationale behind Changes
This makes ADSR more accurate, mainly for the exponential decrement mode which was slightly off before.

### Suggested Testing Steps
Affects anything and everything.

### CODE REVIEW

File: pcsx2/SPU2/Mixer.cpp [hunk 1]
```diff
@@ -679,19 +657,13 @@ __forceinline
 	}
 	else
 	{
-		Out.Left = MulShr32(Out.Left, Cores[1].MasterVol.Left.Value);
-		Out.Right = MulShr32(Out.Right, Cores[1].MasterVol.Right.Value);
+		Out.Left = ApplyVolume(Out.Left, Cores[1].MasterVol.Left.Value);
+		Out.Right = ApplyVolume(Out.Right, Cores[1].MasterVol.Right.Value);
```
Comment (Ziemas) [inline pcsx2/SPU2/Mixer.cpp:661]: The previous use of MulShr32 here seems like a bug, properly applying volume here is what causes the big difference in final output volume between this PR and master.

**Comment by Ziemas** [comment]
Pushed a commit that lowers the volume output, not quite to half like before, but maybe enough to not surprise people. See what you think.

**Comment by RedDevilus** [comment]
Ironically, 200% was introduced because final volume was too low at times. Also you have a typo voluem in comments.

**Review by kamfretoz — APPROVED** [review]
I can definitely feel the difference that the PR's volume is just a _tiny_ bit louder, i suppose this should be fine for most people

**Review by Mrlinkwii — APPROVED** [review]

**Review by JordanTheToaster — APPROVED** [review]

**Pull request merged**